src/FluentNHibernate/PersistenceModel.cs

```diff
@@ -77,12 +77,15 @@ public void AddMappingsFromAssembly(Assembly assembly)
 
         public void AddMappingsFromSource(ITypeSource source)
         {
-            source.GetTypes()
+            foreach (object o in source.GetTypes()
                 .Where(x => IsMappingOf<IMappingProvider>(x) ||
                             IsMappingOf<IIndeterminateSubclassMappingProvider>(x) ||
                             IsMappingOf<IExternalComponentMappingProvider>(x) ||
                             IsMappingOf<IFilterDefinition>(x))
-                .Each(Add);
+                .AsParallel().AsOrdered().Select(t => t.InstantiateUsingParameterlessConstructor()).ToList())

```

Comment:
This could be a possible breaking change for applications needing to spare as much as possible the threads count they are using. Of course normally configuration happens at startup, at a time where sparing threads usually matters less. But that may not be always the case.

So, I think we need a way to choose not to use any parallelism for building the mappings. And enabling parallel build should rather be opt-in, to avoid being a possible breaking change needing a code or config change to be reverted for existing applications.